lib/cocoapods-core/source.rb

```diff
@@ -24,6 +24,7 @@ class Source
     #
     def initialize(repo)
       @repo = Pathname(repo).expand_path
+      @name_to_versions_cache = {}
```

Comment:
@segiddins from a look, no I do not see it. The *only* case I can imagine we might need to invalidate the cache is when we perform `update` of the repo in `source.rb`.

However, updating occurs at the beginning before querying for specs.